pkg/analysis/passes/osvscanner/osvscanner.go

```diff
@@ -0,0 +1,193 @@
+			}
+		}
+	}
+	path, err := exec.LookPath("osv-scanner")
+	if err != nil {
+		pass.ReportResult(
+			pass.AnalyzerName,
+			missingOSVScanner,
+			"Binary for osv-scanner not found in PATH", "osv-scanner executable must be in your shell PATH.")
+		return nil, nil
+	} else {
+		missingOSVScanner.Severity = analysis.OK
+		if missingOSVScanner.ReportAll {
+			pass.ReportResult(
+				pass.AnalyzerName,
+				missingOSVScanner,
+				"Binary for osv-scanner was found in PATH", "osv-scanner executable exists in your shell PATH.")
+		}
+	}
+	// exec
+	cmdArgs := []string{"--json", "--lockfile", lockFile}
+	data, err := exec.Command(path, cmdArgs...).Output()
+	// error output is expected from osv-scanner
+	if err != nil {
+		if len(string(err.Error())) == 0 {
+			pass.ReportResult(
+				pass.AnalyzerName,
+				scanningFailure,
+				"error running osv-scanner",
+				fmt.Sprintf("osv-scanner found, but failed to run: %s", err))
+			return nil, nil
+		}
+	} else {
+		scanningFailure.Severity = analysis.OK
+		pass.ReportResult(
+			pass.AnalyzerName,
+			scanningFailure,
+			"osv-scanner successfully ran",
+			"osv-scanner successfully ran and has output")
+	}
+
+	// deserialize json output, detect CRITICAL severity
+
+	var objmap OSVJsonOutput
+	if err := json.Unmarshal(data, &objmap); err != nil {
+		pass.ReportResult(
+			pass.AnalyzerName,
+			scanningFailure,
+			"osv-scanner output not recognized",
+			fmt.Sprintf("osv-scanner output could not be parsed: %s", err))
+		return nil, nil
+	}
+
+	// no results means no issues currently reported
+	if len(objmap.Results) == 0 {
+		return nil, nil
+	}
+
+	criticalSeverityCount := 0
+	highSeverityCount := 0
+	moderateSeverityCount := 0
+	lowSeverityCount := 0
+
+	// iterate over results
+	for _, result := range objmap.Results {
+		for _, aPackage := range result.Packages {
+			for _, aVulnerability := range aPackage.Vulnerabilities {
+				aliases := strings.Join(aVulnerability.Aliases, " ")
+				message := fmt.Sprintf("SEVERITY: %s in package %s, vulnerable to %s", aVulnerability.DatabaseSpecific.Severity, aPackage.Package.Name, aliases)
+				switch aVulnerability.DatabaseSpecific.Severity {
+				case SeverityCritical:
+					pass.ReportResult(
+						pass.AnalyzerName,
+						osvScannerCriticalSeverityDetected,
+						"osv-scanner detected a critical severity issue",
+						message)
+					criticalSeverityCount++
+				case SeverityHigh:
+					pass.ReportResult(
+						pass.AnalyzerName,
+						osvScannerHighSeverityDetected,
+						"osv-scanner detected a high severity issue",
+						message)
+					highSeverityCount++
+				case SeverityModerate:
+					pass.ReportResult(
+						pass.AnalyzerName,
+						osvScannerModerateSeverityDetected,
+						"osv-scanner detected a moderate severity issue",
+						message)
+					moderateSeverityCount++
+				case SeverityLow:
+					pass.ReportResult(
```

Comment:
based on my own experience with security scanners we'll get a massive spam of low and mid security reports that will burry everything else in them.

I'd suggest we create a new config option where we can enable these reports in particular instead of reporting them by default.

Reply:
right - after seeing the output from a variety of test files, the amount of sub-critical items can be long (and not something important enough to mitigate)